crates/wash/src/cli/mod.rs

```diff
@@ -259,17 +258,91 @@ impl CommandOutput {
     }
 }
 
+/// Non-generic copy of etcetera::AppStrategy trait to make it dyn-compatible
+pub trait DirectoryStrategy: std::fmt::Debug + Send + Sync {
```

Comment:
I modeled the methods to match one to one the ones from `AppStrategy`, with the exception to avoid generic methods to make it dyn-compatible. This avoids having to change the code at the call-sites.